### PR TITLE
removes metal walls, adds rusted walls

### PR DESCRIPTION
**Pull request by Peptide90**
## About the PR 
As per title, it's been a long time coming but metal walls are getting removed now. There has always been an alternative available so if anyone has any missing prototype issues they can edit the yaml to use solid walls.

Also adds rusted walls for that rough look. They have 100 less health than a normal wall and can be fixed to normal walls with a welder.

To do for someone is adding the wall rot event that makes it spread. Will make explosions more deadly to the station so worth fixing. No variant for reinforced walls, head canon is plasteel doesn't rust but that's up to someone else to do.
**Screenshots**

Content Client_EFIh69HqDm

**Changelog**

:cl:
- add: Added rusted metal walls! Use a welder to remove the rust. They're weaker so be careful.

### CODE REVIEW

File: Resources/Prototypes/Entities/Structures/Walls/walls.yml [hunk 1]
```diff
@@ -585,7 +555,45 @@
...
+- type: entity
+  parent: WallBase
+  id: WallSolidRust
+  name: solid wall
+  suffix: rusted
+  components:
+  - type: Sprite
+    sprite: Structures/Walls/solidrust.rsi
+  - type: Icon
+    sprite: Structures/Walls/solidrust.rsi
```
Comment (metalgearsloth) [inline Resources/Prototypes/Entities/Structures/Walls/walls.yml:568]: Does this need an icon?

**Review by metalgearsloth — CHANGES_REQUESTED** [review]
> can be fixed to normal walls with a crowbar (can't use a welder like in SS13 as it deconstructs them)

Basic implementation stuff like this shouldn't come up in design IMO.

File: Resources/Prototypes/Entities/Structures/Walls/walls.yml (hunk 1 above)
Comment (Peptide90) [reply Resources/Prototypes/Entities/Structures/Walls/walls.yml:568]: More or less copied the entity it replaced. Is it not for wall smoothing?

**Comment by Peptide90** [comment]
> > can be fixed to normal walls with a crowbar (can't use a welder like in SS13 as it deconstructs them)
> 
> Basic implementation stuff like this shouldn't come up in design IMO.

What's the potential fix? Just not have rusted walls or? Others wanted to give Janitor a wire brush but I also disagree with adding a new entity with a single purpose.

File: Resources/Prototypes/Entities/Structures/Walls/walls.yml (hunk 1 above)
Comment (metalgearsloth) [reply Resources/Prototypes/Entities/Structures/Walls/walls.yml:572]: Nope, that's IconSmoothing. Icon is deprecated in most instances as appearance visualization is run on the prototype so 99% it's unnecessary now.

**Comment by metalgearsloth** [comment]
> > > can be fixed to normal walls with a crowbar (can't use a welder like in SS13 as it deconstructs them)
> > 
> > 
> > Basic implementation stuff like this shouldn't come up in design IMO.
> 
> What's the potential fix? Just not have rusted walls or? Others wanted to give Janitor a wire brush but I also disagree with adding a new entity with a single purpose.

Having construction run before welding actions probably?

**Comment by moonheart08** [comment]
Just require the wall be unrusted before decon?

**Comment by Zumorica** [comment]
> Having construction run before welding actions probably?

Yeah, this is the proper fix for any weird construction interactions.
Construction will only run and handle the event if one of the currently available steps involve the tool/item you interacted with.

**Comment by Peptide90** [comment]
Yeah I didn’t understand the original comment tbh. Just realised deconstruction didn’t make much sense anyway so just removed it so you gotta clean it first.

File: Resources/Prototypes/Entities/Structures/Walls/walls.yml [hunk 2]
```diff
@@ -241,36 +241,6 @@
     key: walls
     base: ice
 
-- type: entity
-  parent: WallBase
-  id: WallMetal
```
Comment (wrexbe) [inline Resources/Prototypes/Entities/Structures/Walls/walls.yml:246]: You removed normal walls?

**Review by wrexbe — CHANGES_REQUESTED** [review]

File: Resources/Prototypes/Entities/Structures/Walls/walls.yml (hunk 2 above)
Comment (Peptide90) [reply Resources/Prototypes/Entities/Structures/Walls/walls.yml:246]: Normal walls are solid walls, metal walls have been deprecated forever

Comment (wrexbe) [reply Resources/Prototypes/Entities/Structures/Walls/walls.yml:246]: Your gonna need to fix the maps if you want to kill this

Comment (Peptide90) [reply Resources/Prototypes/Entities/Structures/Walls/walls.yml:246]: There wasn't any issues at time of PR. If it's conflicting now that's the price of keeping up with map merges :P will check.

**Review by metalgearsloth — APPROVED** [review]

**Pull request merged**